### PR TITLE
Fix generator creation name

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

We cannot pass a name to a generator creation like for the other elements creation

**What is the new behavior (if this is a feature change)?**
We can pass a name

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: java/src/main/java/com/powsybl/dataframe/network/adders/GeneratorDataframeAdder.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ public class GeneratorDataframeAdder extends AbstractSimpleAdder {
 
     private static final List<SeriesMetadata> METADATA = List.of(
             SeriesMetadata.stringIndex("id"),
+            SeriesMetadata.strings("name"),
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/dataframe/network/adders/GeneratorDataframeAdder.java:39]: It needs to be added in the adder

Comment (geofjamg) [reply java/src/main/java/com/powsybl/dataframe/network/adders/GeneratorDataframeAdder.java:39]: ?

Comment (EtienneLt) [reply java/src/main/java/com/powsybl/dataframe/network/adders/GeneratorDataframeAdder.java:39]: in createAdder I think a  applyIfPresent(name, row, adder::setName); is missing, or maybe I am wrong

Comment (geofjamg) [reply java/src/main/java/com/powsybl/dataframe/network/adders/GeneratorDataframeAdder.java:39]: it is already in the base class

**Review by EtienneLt — APPROVED** [review]

**Pull request merged**